libcentrifugo/engine/engineredis/engine.go

```diff
@@ -356,38 +361,133 @@ func getConfig(getter config.Getter) *Config {
 		}
 	}
 
+	password := getter.GetString("redis_password")
+	db := getter.GetString("redis_db")
+
 	if len(sentinelAddrs) > 0 && masterName == "" {
 		logger.FATAL.Fatalln("Redis master name required when Sentinel used")
 	}
 
-	conf := &Config{
-		Host:           getter.GetString("redis_host"),
-		Port:           getter.GetString("redis_port"),
-		Password:       getter.GetString("redis_password"),
-		DB:             getter.GetString("redis_db"),
-		URL:            getter.GetString("redis_url"),
-		PoolSize:       getter.GetInt("redis_pool"),
-		API:            getter.GetBool("redis_api"),
-		NumAPIShards:   getter.GetInt("redis_api_num_shards"),
-		Prefix:         getter.GetString("redis_prefix"),
-		MasterName:     masterName,
-		SentinelAddrs:  sentinelAddrs,
-		ConnectTimeout: time.Duration(getter.GetInt("redis_connect_timeout")) * time.Second,
-		ReadTimeout:    time.Duration(getter.GetInt("redis_read_timeout")) * time.Second,
-		WriteTimeout:   time.Duration(getter.GetInt("redis_write_timeout")) * time.Second,
+	var shardConfigs []*Config
+
+	if len(hosts) < numShards {
+		newHosts := make([]string, numShards)
+		for i := 0; i < numShards; i++ {
+			newHosts[i] = hosts[0]
```

Comment:
This seems like an odd decisions, I'd just fail or try to distribute shards evenly between the hosts rather than put all the remaining shards on the first host...